sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -1115,9 +1126,38 @@ object SQLContext {
             })
         }
     }
-    def createConverter(cls: Class[_], dataType: DataType): Any => Any = dataType match {
-      case struct: StructType => createStructConverter(cls, struct.map(_.dataType))
-      case _ => CatalystTypeConverters.createToCatalystConverter(dataType)
+    def createConverter(t: Type, dataType: DataType): Any => Any = (t, dataType) match {
```

Comment:
BTW, how about we put this method in `CatalystTypeConverters`? Looks it is a Catalyst converter for beans. Few Java types like `java.lang.Iterable`, `java.math.BigDecimal` and `java.math.BigInteger` are being handled there.

Reply:
I'm okay to move this to `CatalystTypeConverters `, but note that unfortunately seems like `CatalystTypeConverters` doesn't work properly with nested beans as we are trying to support it here.

Reply:
Yea .. was just thinking of moving this func to there .. looks ugly that this file getting long.

Reply:
I took a quick look at `CatalystTypeConverters` and I believe there would be a problem in not being able to reliably distinguish Java beans from other arbitrary classes. We might use setters or set fields directly to objects which would not be prepared for such manipulation, potentially creating hard to find errors. This method already assumes a Java bean so that problem is not present here. Isn't that so?